kernel/aster-nix/src/syscall/access.rs

```diff
@@ -1,11 +1,90 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use super::{constants::*, SyscallReturn};
-use crate::{prelude::*, util::read_cstring_from_user};
+use super::SyscallReturn;
+use crate::{
+    fs::{
+        file_table::FileDesc,
+        fs_resolver::{FsPath, AT_FDCWD},
+        utils::PATH_MAX,
+    },
+    prelude::*,
+    util::read_cstring_from_user,
+};
+
+pub fn sys_faccessat(dirfd: FileDesc, path_ptr: Vaddr, mode: u16) -> Result<SyscallReturn> {
+    debug!(
+        "faccessat: dirfd = {}, path_ptr = {:#x}, mode = {:o}",
+        dirfd, path_ptr, mode
+    );
+
+    do_faccessat(dirfd, path_ptr, mode, 0)
+}
+
+pub fn sys_access(path_ptr: Vaddr, mode: u16) -> Result<SyscallReturn> {
+    debug!("access: path_ptr = {:#x}, mode = {:o}", path_ptr, mode);
+
+    do_faccessat(AT_FDCWD, path_ptr, mode, 0)
+}
+
+bitflags! {
+    struct FaccessatFlags: i32 {
+        const AT_EACCESS = 0x200;
+        const AT_SYMLINK_NOFOLLOW = 0x100;
+        const AT_EMPTY_PATH = 0x1000;
```

Comment:
Should not support the `AT_EMPTY_PATH` flag according to https://man7.org/linux/man-pages/man2/access.2.html

```
       flags is constructed by ORing together zero or more of the
       following values:

       AT_EACCESS
              Perform access checks using the effective user and group
              IDs.  By default, faccessat() uses the real IDs (like
              access()).

       AT_SYMLINK_NOFOLLOW
              If pathname is a symbolic link, do not dereference it:
              instead return information about the link itself.
```

Reply:
This is another implementation detail on Linux. I don't know why they keep `AT_EMPTY_PATH`, so I just follow their design in case that some internal features rely on it.